### ISSUE TITLE
Uninitialized readonly property $id, ManyToMany, cascade delete

### ISSUE THREAD

**Issue by khorsky** [phpstan/phpstan-doctrine#660]
I believe I've found a false positive error of PHPStan for this piece of code:

```php
#[ORM\Entity]
#[ORM\Table(name: 'users')]
class User {

  #[ORM\Id]
  #[ORM\Column(type: 'integer')]
  #[ORM\GeneratedValue]
  private readonly int $id;

  /** @var \Doctrine\Common\Collections\Collection<int, \App\User\UserRole> */
  #[ORM\ManyToMany(targetEntity: UserRole::class, cascade: ['delete'])]
  #[ORM\JoinTable(name: 'users_roles')]
  #[ORM\JoinColumn(name: 'user')]
  #[ORM\InverseJoinColumn(name: 'role')]
  private Collection $roles;

}
```
When I remove `cascade: ['delete']`, PHPStan is satisfied. With `cascade: ['delete']` I get **Class App\User\User has an uninitialized readonly property $id. Assign it in the constructor** error.

phpstan/phpstan 2.1.14
phpstan/phpstan-doctrine 2.0.3

I'm not sure how to link to a reproducible example on PHPStan playground with phpstan/phpstan-doctrine plugin.

**Comment by khorsky** [comment]
**I'm an idiot**, I've to use cascade: ['remove']. Sorry!
